crates/aptos-api-tester/src/tests/nft_transfer.rs

```diff
@@ -0,0 +1,578 @@
+        FAIL_WRONG_TOKEN_DATA, OFFER_TOKEN, SETUP,
+    },
+    time_fn,
+    token_client::{
+        CollectionData, CollectionMutabilityConfig, RoyaltyOptions, TokenClient, TokenData,
+        TokenMutabilityConfig,
+    },
+    utils::{
+        check_balance, create_and_fund_account, emit_step_metrics, get_client, get_faucet_client,
+        NetworkName, TestFailure, TestName,
+    },
+};
+use aptos_api_types::U64;
+use aptos_logger::info;
+use aptos_rest_client::Client;
+use aptos_sdk::types::LocalAccount;
+use aptos_types::account_address::AccountAddress;
+
+const COLLECTION_NAME: &str = "test collection";
+const TOKEN_NAME: &str = "test token";
+const TOKEN_SUPPLY: u64 = 10;
+const OFFER_AMOUNT: u64 = 2;
+
+/// Tests nft transfer. Checks that:
+///   - collection data exists
+///   - token data exists
+///   - token balance reflects transferred amount
+pub async fn test(network_name: NetworkName, run_id: &str) -> Result<(), TestFailure> {
+    // setup
+    let (client, mut account, mut receiver) = emit_step_metrics(
+        time_fn!(setup, network_name),
+        TestName::NftTransfer,
+        SETUP,
+        network_name,
+        run_id,
+    )?;
+    let token_client = TokenClient::new(&client);
+
+    // check account data persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::address_address,
+            CHECK_ACCOUNT_DATA,
+            check_account_data,
+            &client,
+            account.address(),
+            receiver.address()
+        ),
+        TestName::NftTransfer,
+        CHECK_ACCOUNT_DATA,
+        network_name,
+        run_id,
+    )?;
+
+    // create collection
+    emit_step_metrics(
+        time_fn!(create_collection, &client, &token_client, &mut account),
+        TestName::NftTransfer,
+        CREATE_COLLECTION,
+        network_name,
+        run_id,
+    )?;
+
+    // check collection metadata persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::token_address,
+            CHECK_COLLECTION_METADATA,
+            check_collection_metadata,
+            &token_client,
+            account.address()
+        ),
+        TestName::NftTransfer,
+        CHECK_COLLECTION_METADATA,
+        network_name,
+        run_id,
+    )?;
+
+    // create token
+    emit_step_metrics(
+        time_fn!(create_token, &client, &token_client, &mut account),
+        TestName::NftTransfer,
+        CREATE_TOKEN,
+        network_name,
+        run_id,
+    )?;
+
+    // check token metadata persistently
+    emit_step_metrics(
+        time_fn!(
+            persistent_check::token_address,
+            CHECK_TOKEN_METADATA,
+            check_token_metadata,
+            &token_client,
+            account.address()
+        ),
+        TestName::NftTransfer,
+        CHECK_TOKEN_METADATA,
+        network_name,
+        run_id,
+    )?;
+
+    // offer token
+    emit_step_metrics(
+        time_fn!(
```

Comment:
I wonder, could we move the `time_fn` invocation inside `emit_step_metrics` somehow? I haven't thought very hard about it so I'm not sure if it's possible or not, but if it is, it'd make these a bit more readable.